openslides_backend/restrictions/restrictions.py

```diff
@@ -0,0 +1,148 @@
+from typing import Any, Callable, Dict, Type
+
+import fastjsonschema  # type: ignore
+from fastjsonschema import JsonSchemaException  # type: ignore
+
+from ..shared.exceptions import RestrictionException
+from ..shared.interfaces import LoggingModule, Services
+from ..shared.patterns import KEYSEPARATOR, Collection, FullQualifiedField
+from ..shared.schema import schema_version
+from .restrictions_interface import Payload, RestrictionResponse
+
+Restriction = Any  # TODO
+
+
+def prepare_restrictions_map() -> None:
+    """
+    This function just imports all restrictions modules so that the restrictions
+    are recognized by the system and the register decorator can do its work.
+
+    New modules have to be added here.
+    """
+    # from . import meeting, topic  # type: ignore # noqa
+    pass
+
+
+restrictions_map: Dict[Collection, Type[Restriction]] = {}
+
+
+def register_restriction(
+    name: Collection,
+) -> Callable[[Type[Restriction]], Type[Restriction]]:
+    """
+    Decorator to be used for restriction classes. Registers the class so that it
+    can be found by the handler.
+    """
+
+    def wrapper(restriction: Type[Restriction]) -> Type[Restriction]:
+        restrictions_map[name] = restriction
+        return restriction
+
+    return wrapper
+
+
+prepare_restrictions_map()
+
+
+payload_schema = fastjsonschema.compile(
+    {
+        "$schema": schema_version,
+        "title": "Schema for restrictions API",
+        "description": "An array of restriction blobs, i. e. bundles of user_id and full qualified fields.",
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "user_id": {
+                    "description": "Id of the user the given fields should be restricted for.",
+                    "type": "number",
+                    "mininmum": 0,
+                },
+                "fqfields": {
```

Comment:
I would suggest, that you call this "keys" since it is not only about the fields but about all of the key. But I can live with fqfields in this place